lib/src/collection/treeset.dart

```diff
@@ -84,56 +85,61 @@ enum TreeSearch {
 /// A node in the [TreeSet].
 abstract class _TreeNode<V> {
   /// TreeNodes are always allocated as leafs.
-  _TreeNode({this.object});
+  _TreeNode({required this.object});
 
   _TreeNode<V> get left;
+  bool get hasLeft;
+
   _TreeNode<V> get right;
+  bool get hasRight;
 
   // TODO(codefu): Remove need for [parent]; this is just an implementation
   // note.
   _TreeNode<V> get parent;
+  bool get hasParent;
+
   V object;
 
   /// Return the minimum node for the subtree
   _TreeNode<V> get minimumNode {
-    var node = this;
-    while (node.left != null) {
+    _TreeNode<V> node = this;
```

Comment:
was the type on the left required? I thought it would get the type form "this"?

Reply:
same in a couple of places below. I'm OK with "yeah, NNBD requires the type so it can't be null".

Reply:
I can change it back to var. Added the type since during migration I had loosened it to `_TreeNode<V>?` while before I addded hasLeft/hasRight and a non-null return type on left/right. var is fine here.

Reply:
Done.